Use `@cala/koa-middleware` npm package

scotttrinh opened the pull request:
The three middlewares from this project were moved to ca-la/koa-middleware and are being imported here.

Added a smoke test just to make sure that the middlewares were still working as expected after the refactor.

scotttrinh committed:
Initial refactor to @cala/koa-middleware

scotttrinh committed:
Use @cala/koa-middleware and add tests

scotttrinh commented on package.json:
```diff
@@ -29,23 +29,29 @@
     "svg2png": "4.1.1",
     "svgbundler": "1.3.2",
-    "typescript": "2.5.2"
+    "typescript": "^2.9.2"
```
@dylanpyle 

I had to bump this to get `@types/sinon` to compile. Should I go through and change the other dependencies to `^` in this PR?

dylanpyle replied:
@scotttrinh Yeah, that sounds like a plan to me

dylanpyle commented on test.ts:
```diff
@@ -0,0 +1,30 @@
+    .expect('X-Powered-By', /svgthumb-api/)
+    .expect('S3 URL')
+    .end((error: Error): void => {
+      consoleStub.restore();
```
I think we need to restore `awsUploadStub` too (or maybe preferable: use a [sandbox](http://sinonjs.org/releases/v1.17.7/sandbox/) and restore the whole set at once)

scotttrinh replied:
Using `sinon@6.1.3` now, so apparently:

> Since sinon@5.0.0, the sinon object is a default sandbox. Unless you have a very advanced setup or need a special configuration, you probably want to just use that one.

Cool!

dylanpyle commented on package.json:
```diff
@@ -4,7 +4,7 @@
   "main": "index.js",
   "scripts": {
-    "test": "make lint && make build",
+    "test": "make lint && make test && make build",
```
Do we ever use `npm test`?

scotttrinh replied:
I was using it locally to kick off linting and testing at the command line. Seems like we can either just remove the whole thing or at the very least add the new test to the `npm test` script, yeah? I'm fine with either.

dylanpyle replied:
🤷‍♂️ if you use this, this is great with me

dylanpyle reviewed (changes requested):
Great stuff. Looks good after the stub thing, the other comments are just questions

scotttrinh committed:
Use sandbox to restore all stubs

scotttrinh committed:
Use default sandbox in sinon

scotttrinh committed:
Use ^ semver range for dependencies

Aligns with how we're declaring dependencies in the API. Since we have a
package-lock file checked in, no need to specify exact versions.

scotttrinh committed:
Skip importing all types in compilation step

dylanpyle reviewed (approved):
sorry for the delay! i bet this threw off our metrics!!
